core/io/resource_saver.cpp

```diff
@@ -146,6 +153,23 @@ Error ResourceSaver::save(const Ref<Resource> &p_resource, const String &p_path,
 	return err;
 }
 
+Error ResourceSaver::set_uid(const String &p_path, ResourceUID::ID p_uid) {
+	String path = p_path;
+
+	ERR_FAIL_COND_V_MSG(path.is_empty(), ERR_INVALID_PARAMETER, "Can't update UUID to empty path. Provide non-empty path.");
```

Comment:
Some tweaks (main change prompting this suggestion was fixing "UUID" to "UID", since we made sure to only use the latter in the API).
```suggestion
	ERR_FAIL_COND_V_MSG(path.is_empty(), ERR_INVALID_PARAMETER, "Can't update UID for an empty path. Provide a valid resource path.");
```